app/src/main/java/lt/vilnius/tvarkau/fragments/ProblemsMapFragment.java

```diff
@@ -1,14 +1,155 @@
 package lt.vilnius.tvarkau.fragments;
 
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.support.v4.content.ContextCompat;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
 import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.BitmapDescriptor;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import java.util.HashMap;
+
+import lt.vilnius.tvarkau.ProblemDetailActivity;
+import lt.vilnius.tvarkau.R;
+import lt.vilnius.tvarkau.entity.Problem;
+import lt.vilnius.tvarkau.factory.DummyProblems;
+import lt.vilnius.tvarkau.utils.PermissionUtils;
+import lt.vilnius.tvarkau.views.adapters.MapsInfoWindowAdapter;
 
 /**
  * Created by Karolis Vycius on 2016-01-13.
  */
-public class ProblemsMapFragment extends SupportMapFragment {
+public class ProblemsMapFragment extends SupportMapFragment
+        implements OnMapReadyCallback,
+        GoogleMap.OnInfoWindowClickListener,
+        GoogleMap.OnInfoWindowCloseListener,
+        GoogleMap.OnMarkerClickListener {
+
+    protected static final LatLng VILNIUS_LAT_LNG = new LatLng(54.687157, 25.279652);
+    protected static final int GPS_PERMISSION_REQUEST_CODE = 11;
+
+    protected GoogleMap googleMap;
+
+    protected BitmapDescriptor inProgressMarker;
+    protected BitmapDescriptor doneMarker;
+
+    protected HashMap<String, Problem> problemHashMap = new HashMap<>();
 
     public static ProblemsMapFragment getInstance() {
         return new ProblemsMapFragment();
     }
 
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        setMarkerResources();
+
+        getMapAsync(this);
+    }
+
+    private void setMarkerResources() {
+        inProgressMarker = BitmapDescriptorFactory.fromResource(R.drawable.ic_pin_drop_blue);
+        doneMarker = BitmapDescriptorFactory.fromResource(R.drawable.ic_pin_drop_green);
+    }
+
+    protected void requestGPSPermission() {
```

Comment:
We already have method for requesting permission. This places can be changed to just:
```java
        if(PermissionUtils.verifyAndRequestPermissions(getActivity(), GPS_PERMISSION_REQUEST_CODE,
                Manifest.permission.ACCESS_FINE_LOCATION)) {
            googleMap.setMyLocationEnabled(true);
        }
```
Although if we do this Android studio will give us a warning:
![screenshot from 2016-02-01 13-36-02](https://cloud.githubusercontent.com/assets/3719141/12716390/d7235030-c8e8-11e5-96a3-836db3364350.png)
I don't know what is the best way to avoid it.